frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -965,34 +968,45 @@ pub mod pallet {
             GeneralEraInfo::<T>::insert(era, era_info);
         }
 
-        /// This helper returns `ContractStakeInfo` for given era if possible or latest stored data
-        /// or finally default value if storage have no data for it.
+        /// Used to copy all `ContractStakeInfo` from the ending era over to the next era.
+        /// This is the most primitive solution since it scales with number of dApps.
+        /// It is possible to provide a hybrid solution which allows laziness but also prevents
+        /// a situation where we don't have access to the required data.
+        fn advance_staking_info(current_era: EraIndex) -> u64 {
+            let next_era = current_era + 1;
+
+            let mut consumed_weight = 0;
+
+            for (contract_id, dapp_info) in RegisteredDapps::<T>::iter() {
```

Comment:
Whats about to iterate over `ContractEraStake` for current era? It will save some weight.

Reply:
So, the cleanup unregistered dapps could be an issue in that case.

Reply:
But unregistered dapps should have staking info for user unstake, didn't? We can cleanup data when stake becomes `0`.

Reply:
The key format is: <contract_id, era> so we cannot iterate over prefix, unfortunately.

We cannot clean data when stake becomes `0` since we need to know when it became `0`.
It's important we keep it in storage (up to history depth).

Reply:
Oh, I misunderstood - you meant `unregistered` dapps.

I was thinking of handling this in a bit different way as a part of future improvement - add a cleanup extrinsic call that will remove all data related to a particular dapp